cmd/run_python.go

```diff
@@ -0,0 +1,83 @@
+package cmd
+
+import (
+	"context"
+	"fmt"
+	"os"
+	"os/exec"
+	"path/filepath"
+	"strconv"
+	"strings"
+
+	"go.temporal.io/sdk-features/harness/go/cmd"
+	"go.temporal.io/server/common/log/tag"
+)
+
+// RunPythonExternal runs the Python run in an external process. This expects
+// the server to already be started.
+func (r *Runner) RunPythonExternal(ctx context.Context, run *cmd.Run) error {
+	// To do this, if they supplied a custom version we'll make a temporary
+	// directory to run in with a custom pyproject and such. Otherwise we'll just
+	// use this directory.
+
+	runDir := r.rootDir
+	if r.config.Version != "" {
+		// Put pyproject.toml in temp dir
+		var err error
+		if runDir, err = os.MkdirTemp(r.rootDir, "sdk-features-python-test-"); err != nil {
+			return fmt.Errorf("failed creating temp dir: %w", err)
+		}
+		r.createdTempDir = &runDir
+		r.log.Info("Building temporary Python project", tag.NewStringTag("Path", runDir))
+		// Use semantic version or path if it's a path
+		versionStr := strconv.Quote(r.config.Version)
+		if strings.ContainsAny(r.config.Version, "/\\") {
+			// We expect a dist/ directory with a single whl file present
+			wheels, err := filepath.Glob(filepath.Join(r.config.Version, "dist/*.whl"))
+			if err != nil {
+				return fmt.Errorf("failed glob wheel lookup: %w", err)
+			} else if len(wheels) != 1 {
+				return fmt.Errorf("expected single dist wheel, found %v", len(wheels))
+			}
+			versionStr = "{ path = " + strconv.Quote(wheels[0]) + " }"
+		}
+		pyProjectTOML := `
```

Comment:
Wouldn't you rather generate this with a structure that maps to toml? E.g: https://github.com/pelletier/go-toml

Reply:
Nah, doesn't seem necessary for such small use. This is a single, small temp file. Were I to do more advanced TOML mods, definitely.